site/source/docs/contributing/developers_guide.rst

```diff
@@ -95,6 +98,46 @@ automatically when you create a pull request, and they should all pass. If you
 run into trouble with a test failure you can't fix, please let the developers
 know.
 
+Bisecting
+=========
+
+If you find a regression, bisection is often the fastest way to figure out what
+went wrong. This is true not just for finding an actual regression in Emscripten
+that you want to fix, but also if your project stopped working when you upgrade,
+and you need to investigate if it's an Emscripten regression or something else.
+
+The normal ``git bisect`` workflow can work if you just need to bisect a single
+repository. For example, you can bisect only on the emscripten repo if you are on
+a range that all works with the same versions of LLVM and Binaryen (which was
+mentioned earlier, are two separate codebases that are depended on).
+
+If you have a large bisection range, you generally can't bisect a single repo.
+You can still bisect, though! To do that you need the emsdk and to understand
+how the
+`release process <https://github.com/emscripten-core/emscripten/blob/master/docs/process.md#release-processes>`_
+works for all the repos together. The key "trick" is that
+
+::
+
+     emsdk install tot
+
+can install an arbitrary build of emscripten: it installs the one identified
+in ``emscripten-releases-tot.txt``. You can therefore bisect on the
+`releases repo <https://chromium.googlesource.com/emscripten-releases>`_ which
+has a DEPS file that basically says what version of all the various repos are
+in which release. In each bisection step on that repo you get the git hash in
```

Comment:
in a given release?

Reply:
that repo -> this repo

Reply:
How about: `At each bisection step in this repo the git hash identifies a particular `tot` release (which...`.